runtime/runtime.go

```diff
@@ -567,24 +571,20 @@ func (rt *Runtime) startWatcher(ctx context.Context, paths []string, onReload fu
 }
 
 func (rt *Runtime) readWatcher(ctx context.Context, watcher *fsnotify.Watcher, paths []string, onReload func(time.Duration, error)) {
-	for {
-		select {
-		case evt := <-watcher.Events:
-
-			removalMask := (fsnotify.Remove | fsnotify.Rename)
-			mask := (fsnotify.Create | fsnotify.Write | removalMask)
-			if (evt.Op & mask) != 0 {
-				logrus.WithFields(logrus.Fields{
-					"event": evt.String(),
-				}).Debugf("registered file event")
-				t0 := time.Now()
-				removed := ""
-				if (evt.Op & removalMask) != 0 {
-					removed = evt.Name
-				}
-				err := rt.processWatcherUpdate(ctx, paths, removed)
-				onReload(time.Since(t0), err)
+	for evt := range watcher.Events {
```

Comment:
👍 I like this construct

Reply:
Me too. I'm really impressed linters can spot things like this.